collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -355,42 +388,53 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
     }
 
     private fun quote(text: String) = "\"$text\""
-
-    companion object {
-        private const val DATABASE_VERSION = 2
-    }
 }
 
-private class EntitiesDatabaseMigrator :
-    DatabaseMigrator {
+class EntitiesDatabaseMigrator : DatabaseMigrator {
 
     override fun onCreate(db: SQLiteDatabase) {
-        db.execSQL(
-            """
-            CREATE TABLE IF NOT EXISTS ${ListsTable.TABLE_NAME} (
-                    $_ID integer PRIMARY KEY, 
-                    ${ListsTable.COLUMN_NAME} text NOT NULL,
-                    ${ListsTable.COLUMN_HASH} text
-            );
-            """.trimIndent()
-        )
+        createDbForVersion(db, DATABASE_VERSION)
     }
 
     override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int) {
-        dropAllTablesFromDB(db)
+        if (oldVersion == 1) {
+            throw IllegalStateException("Cannot upgrade from this beta version. Please reinstall Collect!")
```

Comment:
Can't we just remove the table and let it create again in the next step instead?

Reply:
That would clear the DB of offline entities meaning that data would be gone for good which I don't think we should do without warning the user. By throwing an exception here, they get a crash on launch so they know that they are in a broken situation.

Reply:
But they can't do anything to fix that state, right? Downgrading isn't supported.